src/ex15_js-oop/task-01.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+function Candy(brand, weight) {
+  this.brand = brand;
+  this.weight = weight;
+}
+function SugarCandy(brand, weight, type) {
+  Candy.call(this, brand, weight);
+  this.type = type;
+}
+function ChocolateCandy(brand, weight, type) {
+  Candy.call(this, brand, weight);
+  this.type = type;
+}
+SugarCandy.prototype = Object.create(Candy.prototype);
+SugarCandy.prototype.constructor = SugarCandy;
+ChocolateCandy.prototype = Object.create(Candy.prototype);
+ChocolateCandy.prototype.constructor = ChocolateCandy;
+function Gift() {
+  this.candies = [];
+}
+Gift.prototype.addCandy = function(candy, qty) {
+  for (let i = 0; i < qty; i++) {
+    this.candies.push(candy);
+  }
+  return this;
+}
+Gift.prototype.getTotalWeight = function() {
+  let totalWeight = 0;
+  this.candies.forEach(function(candy) { totalWeight += candy.weight });
+  return totalWeight.toFixed(3);
+}
+Gift.prototype.sortCandies = function(byProp, selectedType) {
+  switch(byProp) {
+    case 'brand':
+    default:
+      return this.candies.sort((candy1, candy2) => {
```

Comment:
if there is a comparator u can use `a,b` names for calling arguments
its ok

Reply:
I think u can refactor that like 
``
this.candies.sort((a, b) => {
const lowerA = a.brand.toLowerCase(),
          lowerB = b.brand.toLowerCase();

return lowerA < lowerB ? -1 : 1
``

Reply:
Done the refactoring. Thank you!